src/System.Linq.Parallel/tests/Helpers/Functions.cs

```diff
@@ -46,6 +46,16 @@ public static void AssertAlreadyCanceled(Action<CancellationToken> query)
             Assert.Equal(cs.Token, oce.CancellationToken);
         }
 
+        public static void AssertEventuallyCanceled(Action<CancellationToken, Action> query)
+        {
+            CancellationTokenSource cs = new CancellationTokenSource();
+            int countDown = 4;
```

Comment:
Should this be an argument rather than hardcoded?  This places a requirement on the sources coming in.